feature/security/gnsi/credentialz/tests/host_certificates/host_certificates_test.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host_certificates_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openconfig/ondatra/gnmi"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	dutPublicKeyFilename     = "dut.pub"
+	dutCertFilename          = "dut-cert.pub"
+	hostCertificateVersion   = "v1.0"
+	hostCertificateCreatedOn = 1705962293
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func prepareCaAndHostKeys(t *testing.T, dir string) {
+	caCmd := exec.Command(
+		"ssh-keygen",
+		"-t", "ed25519",
+		"-f", "dut",
+		"-C", "dut",
+		"-q", "-N", "", // quiet, empty passphrase
+	)
+	caCmd.Dir = dir
+
+	err := caCmd.Run()
+	if err != nil {
+		t.Fatalf("failed generating ca key pair, error: %s", err)
+	}
+
+	dutCmd := exec.Command(
+		"ssh-keygen",
+		"-t", "ed25519",
+		"-f", "ca",
```

Comment:
Why is it the dutCmd has -f "ca" and the caCmd has -f "dut"? Seems backwards.

Reply:
Fixed